alpakaConfig.cmake

```diff
@@ -122,7 +122,13 @@ ENDIF()
 #-------------------------------------------------------------------------------
 # Check supported compilers.
 IF(CMAKE_CXX_COMPILER_ID MATCHES "Clang" AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS 4.0)
```

Comment:
I think this check is probably just too strict.

Reply:
@BenjaminW3 before we change all of the above, do we even have problems with clang 3.9?

Reply:
We have removed some special casing for clang 3.x 3 months ago. As far as I remember it was about OpenMP support, sanitizers and native CUDA compilation. It could be that we still can compile most user code with clang 3.9 or 3.8 but we simply do not have the possibility to test all those combinations in CI.

Reply:
I am specifically asking about clang 3.9 which is still in wide use and if we test for clang 3.9 we could skip all of the corner-cases below.

Reply:
Clang 3.9 as native CUDA compiler does not work without the workarounds we explicitly removed. If I remember correctly, it did not have support for cuRand. I do not want to re-add those workarounds and it would not get us anything to support this combination.
Clang 3.9 as a host compiler for NVCC might work but we already support the last 3 clang releases of the last 1,5 years. When the current 0.4.0 version gets released there will most probably already be clang 7.

Reply:
Ok, good with me. I just did not remember and the check here did look like a non-cuda accelerator exclusion. But in that case it makes sense.